src/hooks/useQuery.ts

```diff
@@ -77,8 +78,12 @@ export const useQuery = <T = any, V = object>(
 
   useEffect(() => {
     executeQuery();
-    return unsubscribe.current;
-  }, [executeQuery]);
+
+    return () => {
+      isMounted.current = false;
```

Comment:
Small bug here since the effect runs more times than just for mount & unmount

Reply:
Haven't had time to look at this but I'm assuming we'll need a separate effect for managing mounting / unmounting.

Reply:
That makes sense to me, keep the effects separate (which also makes them easier to reason about).

Reply:
The addition of the request key to the effect was to cause a fetch not only on mount but whenever the query changes. I think this is what users would expect but will leave that for a seperate PR.

Reply:
@parkerziegler is the `useQuery.spec.ts` file addition intentional?